crypto/err/err.c

```diff
@@ -588,9 +554,9 @@ static unsigned long get_error_values(int inc, int top, const char **file,
 
 void ERR_error_string_n(unsigned long e, char *buf, size_t len)
 {
-    char lsbuf[64], fsbuf[64], rsbuf[64];
+    char lsbuf[64], rsbuf[64];
```

Comment:
The generated output should not contain any function indicator at all...  but maybe that's too controversial, so the second option is to simply leave that part of the output empty.

Reply:
I think it's useful to know if it's an openssl function ("internal") or an external fuction ("system) which is what the call to ``ERR_func_error_string``` gives you at line 571.

Reply:
That already given with `lsbuf`... I really don't see why the redundancy is needed.

Reply:
There may be applications that parse the colon-separated output and I didn't want to break that.

Reply:
That's why I'm suggesting a second option, i.e. leave the function indicator empty.  In other words, you could make this change at line 579:

``` diff
-     BIO_snprintf(buf, len, "error:%08lX:%s:%s:%s", e, ls, fs, rs);
+     BIO_snprintf(buf, len, "error:%08lX:%s::%s", e, ls, rs);
```

Reply:
(and similar elsewhere in err.c, where applicable)

Reply:
changed.

Reply:
... and now *every single test* in test_errstr fails because they can't parse the empty double colon, leading to failures like:
```
#   at ../test/recipes/02-test_errstr.t line 109.
/home/rsalz/openssl/github/util/shlib_wrap.sh /home/rsalz/openssl/github/apps/openssl errstr 200001d => 0
not ok 66 - (29) ':Illegal seek' == 'Illegal seek'
```

So I don't want to make that change. I will, however, if @t-j-h's hold/-1 is resolved.

Reply:
I'll have a look to see.

Reply:
#9304 should help

Reply:
yes, i'll re-do the change once that gets merged.

Reply:
#9304 is now merged, ball's back in your court :wink:

Reply:
Done.  Nice.  Next question, is what is necessary to clear @t-j-h's -1/hold?  I don't want to spend any more time on this if it's just going to be wasted effort.